src/Servers/Kestrel/Core/src/KestrelServer.cs

```diff
@@ -135,7 +135,30 @@ async Task OnBind(ListenOptions options)
                     }
 
                     var connectionDispatcher = new ConnectionDispatcher(ServiceContext, connectionDelegate);
-                    var transport = await _transportFactory.BindAsync(options.EndPoint).ConfigureAwait(false);
+
+                    IConnectionListenerFactory factory = null;
+                    if (options.Protocols >= HttpProtocols.Http3)
+                    {
+                        foreach (var transportFactory in _transportFactories)
+                        {
+                            if (transportFactory is IMultiplexedConnectionListenerFactory)
+                            {
+                                factory = transportFactory;
+                                break;
+                            }
+                        }
+
+                        if (factory == null)
+                        {
+                            throw new Exception("Quic transport not found when using HTTP/3");
+                        }
+                    }
+                    else
+                    {
+                        factory = _transportFactories.First();
```

Comment:
Don't assume order, how can you check if a transport factory is appropriate for HTTP/1.1 or 2? The absence of IMultiplexedConnectionListenerFactory? Or did you mean Single here?

Reply:
Single.